.github/workflows/docker-publish.yml

```diff
@@ -64,10 +64,13 @@ jobs:
         uses: docker/metadata-action@96383f45573cb7f253c731d3b3ab81c87ef81934 # v5.0.0
         with:
           images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags:
+            type=semver,pattern={{version}},value=v1.0.0
 
       # Build and push Docker image with Buildx (don't push on PR)
       # https://github.com/docker/build-push-action
       - name: Build and push Docker image
+        if: ${{ github.event_name != 'pull_request' && contains(github.event.head_commit, 'Dockerfile')}}
```

Comment:
Uma vez que o objetivo do processo de gerar a imagem é somente se o Dockerifle for alterado, o IF pode ficar no começo das steps, ai nenhuma seria executada e teríamos somente um IF.